### PR TITLE
Add module param to binary add command

### LINKED ISSUE #260: binaries can only be added to the current module
Commands like `fury binary add` do not have a `--module`/`-m` parameter, so you have to change the current module in order to add a binary to it. This isn't ideal.

### PR DESCRIPTION
**Pull request by IvoSaraiva**
The -m option was being supplied in a cli that was not being used neither in the BinariesCtx nor the ParamCtx. Moved it to the ModuleCli.context. 

This should fix #260 for binaries and also for params.

### CODE REVIEW

File: src/module/module.scala [hunk 1]
```diff
@@ -51,6 +51,7 @@ object ModuleCli {
     cli          <- cli.hint(ProjectArg, schema.map(_.projects).getOrElse(Nil))
     optProjectId <- ~schema.flatMap { s => cli.peek(ProjectArg).orElse(s.main) }
     optProject   <- ~schema.flatMap { s => optProjectId.flatMap(s.projects.findBy(_).toOption) }
+    cli          <- cli.hint(ModuleArg, optProject.to[List].flatMap(_.modules))
```
Comment (odisseus) [inline src/module/module.scala]: If this is the right place to generate a hint for the `--module` argument, the value associated with this argument should also be retrieved here, like it is done with `--schema` and `--project`. Consider adding `optModule` as another field to the case class `ModuleCli.Context`.

By the way, doesn't this change enable the hint for `--module` argument even for commands `module add` and `module list`? These two commands don't (and shouldn't) accept that argument.

**Review by odisseus — CHANGES_REQUESTED** [review]

File: src/module/module.scala (hunk 1 above)
Comment (IvoSaraiva) [reply src/module/module.scala]: You're right on the module commands problem. I'll try again.

**Comment by IvoSaraiva** [comment]
What about just updating the MoculeCli.Context.cli in the BinariesCtx and ParamCtx? All binaries and params commands should accept a --module option.

**Review by propensive — APPROVED** [review]

**Comment by odisseus** [comment]
I'm OK with merging this if @propensive approves. But the commits need to be squashed, because there are 5 commits that change just two lines in total.

**Comment by propensive** [comment]
Yes, I just tried it out, and it works great. Nice that the fix was so simple!

**Pull request merged**